src/shogun/mathematics/graph/ops/shogun/Add.h

```diff
@@ -55,7 +56,35 @@ namespace shogun
 					if (CPU_arch->has_avx512f())
 						add_kernel_implementation_avx512f<T>(input1, input2, output, size);
 					else if (CPU_arch->has_avx())
-						add_kernel_implementation_avx<T>(input1, input2, output, size);
+					{
+						size_t i = 0;
+						if (size > 32/sizeof(T))
+						{
+							size_t remainder = size % (32/sizeof(T));
+							for (;i<size-remainder; i+=32/sizeof(T))
+							{
+								if constexpr(std::is_same_v<float, T>)
+								{
+									const auto packet1 = _mm256_loadu_ps(static_cast<const T*>(input1)+i);
+									const auto packet2 = _mm256_loadu_ps(static_cast<const T*>(input2)+i);
```

Comment:
this would be done by the packet implementation

Reply:
This could have been something like `Eigen::internal::ploadu<Packet8f>(static_cast<const T*>(input1)+i)`, if we want to use the eigen implementations